### PR TITLE
Support new class & package from explorer

### LINKED ISSUE #78: Create new class and package from explorer
Is it possible to add a context menu for functions such as `new class` and `new package`?

### LINKED ISSUE #126: The empty package can't display
**### Configuration:**
```
Vscode: 1.29.1
JDK: 1.8.0_191
OS: Win/Mac
Build: 60
```
The empty package can't display in the dependency view, as below:

image

### PR DESCRIPTION
**Pull request by jdneo**
resolve #78

Also fix #126

### CODE REVIEW

File: src/explorerCommands/new.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    // `workspace.applyEdit()` will trigger a workspace file event, and let the
+    // vscode-java extension to handle the type: class, interface or enum.
+    const workspaceEdit: WorkspaceEdit = new WorkspaceEdit();
+    const fsPath: string = getNewFilePath(packageFsPath, className);
+    workspaceEdit.createFile(Uri.file(fsPath));
+    workspace.applyEdit(workspaceEdit);
+}
+
+function getNewFilePath(basePath: string, className: string): string {
+    const indexOfJavaExt: number = className.lastIndexOf(".java");
```
Comment (testforstephen) [inline src/explorerCommands/new.ts:43]: One corner case is "org.java.MyClass", looks like you only need check whether className.endsWith(".java")

Comment (jdneo) [reply src/explorerCommands/new.ts:43]: yes, make sense

File: jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java [hunk 2]
```diff
@@ -440,6 +440,8 @@
                 result.add(child);
             } else if (fragment.getNonJavaResources().length > 0) { // some package has non-java files
                 result.add(fragment);
+            } else if (!fragment.hasSubpackages()) {
+                result.add(fragment);
             }
```
Comment (testforstephen) [inline jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java:445]: looks like you could add the child directly `result.add(child)`. No need of check the conditions.

Comment (jdneo) [reply jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java:445]: That will make every sub packages get rendered in explorer (including the default package) but without any content. For all the empty packages, just show the leaf ones looks much better.

Screen Shot 2020-08-14 at 10 57 59 AM

**Pull request merged**